config/3rd-party/openpmix/pmix_setup_libevent.m4

```diff
@@ -0,0 +1,247 @@
+# -*- shell-script -*-
+#
+# Copyright (c) 2009-2015 Cisco Systems, Inc.  All rights reserved.
+# Copyright (c) 2013      Los Alamos National Security, LLC.  All rights reserved.
+# Copyright (c) 2013-2020 Intel, Inc.  All rights reserved.
+# Copyright (c) 2017-2019 Research Organization for Information Science
+#                         and Technology (RIST).  All rights reserved.
+# Copyright (c) 2020      IBM Corporation.  All rights reserved.
+# Copyright (c) 2020-2021 Amazon.com, Inc. or its affiliates.  All Rights
+#                         reserved.
+# Copyright (c) 2021      Nanook Consulting.  All rights reserved.
+# $COPYRIGHT$
+#
+# Additional copyrights may follow
+#
+# $HEADER$
+#
+
+#
+# We have two modes for building libevent.
+#
+# First is as a co-built libevent.  In this case, PMIx's CPPFLAGS
+# will be set before configure to include the right -Is to pick up
+# libevent headers and LIBS will point to where the .la file for
+# libevent will exist.  When co-building, libevent's configure will be
+# run already, but the library will not yet be built.  It is ok to run
+# any compile-time (not link-time) tests in this mode.  This mode is
+# used when the --with-libevent=cobuild option is specified.
+#
+# Second is an external package.  In this case, all compile and link
+# time tests can be run.  This macro must do any CPPFLAGS/LDFLAGS/LIBS
+# modifications it desires in order to compile and link against
+# libevent.  This mode is used whenever the other modes are not used.
+#
+# PMIX_LIBEVENT_CONFIG()
+# --------------------------------------------------------------------
+AC_DEFUN([PMIX_LIBEVENT_CONFIG],[
+    PMIX_VAR_SCOPE_PUSH([pmix_libevent_source])
+
+    AC_ARG_WITH([libevent],
+                [AS_HELP_STRING([--with-libevent=DIR],
+                                [Search for libevent headers and libraries in DIR ])])
+
+    AC_ARG_WITH([libevent-libdir],
+                [AS_HELP_STRING([--with-libevent-libdir=DIR],
+                                [Search for libevent libraries in DIR ])])
+
+    pmix_libevent_support=1
+
+    AS_IF([test "$with_libevent" = "no"],
+          [AC_MSG_NOTICE([Libevent support disabled by user.])
+           pmix_libevent_support=0])
+
+    # figure out our mode...
+    AS_IF([test "$with_libevent" = "cobuild"],
+          [_PMIX_LIBEVENT_EMBEDDED_MODE()],
+          [_PMIX_LIBEVENT_EXTERNAL([pmix_libevent_support=1],
+                                   [pmix_libevent_support=0])])
+    # Check to see if the above check failed because it conflicted with LSF's libevent.so
+    # This can happen if LSF's library is in the LDFLAGS envar or default search
+    # path. The 'event_getcode4name' function is only defined in LSF's libevent.so and not
+    # in Libevent's libevent.so
+    if test $pmix_libevent_support -eq 0; then
```

Comment:
I added a clarifying comment.  This is ok, because we're not actually running AC_CHECK_LIB on Libevent's -levent, but on LSF's -levent.  In a cobuild, we should never get here anyway, because we should never fail the previous tests in that situation.  But even if we did, this test would still be ok because of which -levent it is looking for.